spdlog-macros/src/pattern/mod.rs

```diff
@@ -19,6 +20,53 @@ pub fn pattern_impl(pattern: Pattern) -> Result<TokenStream> {
     Synthesiser::new(registry).synthesize(pattern.template())
 }
 
+pub fn runtime_pattern_impl(runtime_pattern: RuntimePattern) -> TokenStream {
+    let custom_pattern_names: Vec<_> = runtime_pattern
+        .custom_patterns
+        .0
+        .iter()
+        .map(|(name, _)| name.to_string())
+        .collect();
+    if let Err(err) =
+        spdlog_internal::pattern_parser::check_custom_pattern_names(custom_pattern_names)
+    {
+        panic!("invalid custom pattern names: {}", err);
+    }
+
+    let custom_pattern_register_calls: Vec<_> = runtime_pattern
+        .custom_patterns
+        .0
+        .into_iter()
+        .map(|(name, factory)| {
+            let name_literal = LitStr::new(&name.to_string(), Span::mixed_site());
+            quote! {
+                registry.register_custom(#name_literal, Box::new(|| Box::new(#factory())))
+                    .expect("unexpected panic, please report a bug to spdlog-rs");
```

Comment:
> Mmm, I think we should pass an _array_ of custom patterns into the registry, and storing it in a separated field of `PatternRegistry`. Otherwise there is not much difference compare to the previous `struct RuntimePatternBuilder` approach, it's still building a set of custom patterns at runtime.

The idea is actually more complicated than I expected, so let's just keep it that way.